.github/workflows/register_repositories.py

```diff
@@ -0,0 +1,223 @@
+import asyncio
+import copy
+import json
+import logging
+import os
+from datetime import datetime
+
+import aiohttp
+import ruamel.yaml as yaml
+
+logging.basicConfig(level=logging.INFO)
+
+API_BASE_URL = 'https://api.github.com'
+
+XDEVBOT_MAIN_ENDPOINT = 'http://xdevbot.herokuapp.com/'
+
+
+def parse_line(line, original_config, repos={'remove': [], 'add': []}):
+    config = copy.deepcopy(original_config)
+    add_cmd = '/add-repo'
+    remove_cmd = '/remove-repo'
+    valid_line = add_cmd in line or remove_cmd in line
+    if valid_line:
+        split_on = add_cmd if add_cmd in line else remove_cmd
+        parsed_info = line.split(split_on)[-1].strip().split()[:2]
+        info = {}
+        for item in parsed_info:
+            x = item.strip().split(':')
+            info[x[0]] = x[1]
+
+        if split_on == '/add-repo':
+            if info['repo'] not in set(config[info['campaign']]['repos']):
```

Comment:
This assumes that the specified campaign is "known"/found in the config file. Should we guard against `KeyError` here or trust the user to supply the right campaign?

Reply:
Oh, ugh!  Yes.  We should.  I missed that and just merged the PR.

Do you might doing another PR for this?  It should catch all errors that could come from the command, including unknown campaign or other parsing errors (like no specified repo *or* campaign).